plugins/search/src/components/SearchFilter/SearchFilter.tsx

```diff
@@ -106,6 +106,61 @@ const CheckboxFilter = ({
   );
 };
 
+const TypeFilter = ({
+  values = [],
+  className,
+  name,
+  defaultValue,
+}: Omit<Component, 'defaultValue'> & { defaultValue?: string }) => {
+  const classes = useStyles();
+  const { types, setTypes } = useSearch();
+
+  useEffect(() => {
+    if (defaultValue) {
+      setTypes([defaultValue]);
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  const handleChange = (e: ChangeEvent<{ value: unknown }>) => {
+    const {
+      target: { value },
+    } = e;
+    if (value === '*') {
+      setTypes([]);
+    } else {
+      setTypes([value as string]);
+    }
+  };
+
+  return (
+    <FormControl
+      className={className}
+      variant="filled"
+      fullWidth
+      data-testid="search-typefilter-next"
+    >
+      <InputLabel className={classes.label} margin="dense">
+        {name}
+      </InputLabel>
+      <Select
+        variant="outlined"
+        value={types[0] || '*'}
+        onChange={handleChange}
+      >
+        <MenuItem value="*">
+          <em>All</em>
+        </MenuItem>
+        {values.map((value: string) => (
+          <MenuItem key={value} value={value}>
+            {value}
+          </MenuItem>
+        ))}
+      </Select>
+    </FormControl>
```

Comment:
Backstage is often using a "chip style" when having multiselect elsewhere:

```
<Select
        variant="outlined"
        multiple
        value={types}
        onChange={handleChange}
        renderValue={selected => (
          <div className={classes.chips}>
            {(selected as string[]).map(value => (
              <Chip
                key={value}
                label={value}
                className={classes.chip}
                size="small"
              />
            ))}
          </div>
        )}
      >
        {values.map(value => (
          <MenuItem key={value} value={value}>
            <Checkbox checked={types.indexOf(value) > -1} />
            <ListItemText primary={value} />
          </MenuItem>
        ))}
      </Select>
```

Btw, cool that you added this one, it was on my list of things I wanted to open source 😆

Reply:
Ah, wait, this isn't a multiselect 😅